src/providers/Staking.tsx

```diff
@@ -1,12 +1,19 @@
 import React, { useContext, useMemo } from 'react'
 import { useStaking } from '@hooks/useStaking'
 
-const StakingContext = React.createContext({
-  staking: {},
-  loading: true,
+type State = {
+  stakeManagement: any
+  stakeActions: any
+  loading: any
+}
+
+const StakingContext = React.createContext<State>({
+  stakeManagement: null,
+  stakeActions: null,
+  loading: false,
 })
 
-function StakingProvider({ children }) {
+function StakingProvider({ children }: { children: React.ReactNode }) {
```

Comment:
same here could we keep a coherence with how we define the function signatures with the types? so we all try to do the same way? for example i am defining everything outside the function line

Reply:
Yes, for coherence we should do that, although when its a simple children i dont often do that. But I will do it in Gardens :)